NuGet.Docs/ndocs/API/Visual-Studio-Credential-Providers.md

```diff
@@ -0,0 +1,76 @@
+﻿# NuGet Credential Providers for Visual Studio
+
+The NuGet Visual Studio Extension 3.6+ now supports Credential Providers, which enable NuGet to work seamlessly with authenticated feeds.
+After you install a NuGet Credential Provider for Visual Studio, the NuGet Visual Studio extension will automatically acquire and refresh credentials for authenticated feeds as necessary.
+
+Note: NuGet Credential Providers for Visual Studio only work in Visual Studio (not in dotnet restore or nuget.exe). To learn more about NuGet.exe Credential Providers, check out [this page](NuGet.exe-Credential-Providers.md).
+
+NuGet Credential Providers for Visual Studio must be installed as a regular Visual Studio extension.
+
+## Available NuGet Credential Providers for Visual Studio
+
+There is a single credential provider that is built into the Visual Studio NuGet extension to support Visual Studio Team Services.
+
+The NuGet Visual Studio Extension uses an internal `VsCredentialProviderImporter` which also scans for plug-in credential providers. These plug-in credential providers must be discoverable as a MEF Export of type `IVsCredentialProvider`.
+
+## Creating a NuGet Credential Provider for Visual Studio
+
+The NuGet Visual Studio Extension 3.6+ implements an internal CredentialService that is used to acquire credentials. The CredentialService has a list of built-in and plug-in Credential Providers. Each provider is tried sequentially until credentials are acquired.
+
+During credential acquisition, the credential service will try credential providers in the following order, stopping as soon as credentials are acquired:
+
+a) Credentials will be fetched from NuGet configuration files (using the built-in `SettingsCredentialProvider`).
+b) If the package source is on Visual Studio Team Services, the `VisualStudioAccountProvider` will be used.
+c) All 3rd party plug-in credential providers will be tried in the order of discovery outlined above.
+d) If no credentials have been acquired yet, the user will be prompted for credentials using a standard basic authentication dialog.
+
+To create a NuGet Credential Provider for Visual Studio, create a Visual Studio Extension that exposes a public MEF Export implementing the `IVsCredentialProvider` type, and adheres to the principles outlined below.
+A sample implementation can be found in [our GitHub Samples](https://github.com/NuGet/Samples) repository.
```

Comment:
New line above sample, also call out attention to the sample by creating a new section right after the introduction and put in the link there